### DISCUSSION TITLE
Hi FrankenPHP Team! How to support symlinks inside /app/public folder ? It does not seem to resolve the paths

### DISCUSSION THREAD

**Discussion by alexandreelise** [php/frankenphp#299] [Q&A]
Hi, I tried to create a symlink inside /app/public of FrankenPHP container but the symlink does not seem to be resolved and throws a 404 with a wrong path in the FrankenPHP logs. Is it an alpine linux know issue, is it a caddy server known issue or did I somehow misconfigured something for FrankenPHP. Can you please shed a light on this one? I'll continue to investigate on my part.

**Comment by dunglas** [comment]
Could you show your Dockerfile or reproducer?

**Reply by alexandreelise** [reply]
Dockerfile is private for now. It will be public when I fix this issue.

**Reply by alexandreelise** [reply]
Don't worry. I'll take a deep dive in the realm of Caddy server and FrankenPHP to understand why it does not work for now.

**Reply by alexandreelise** [reply]
I've already started promoting FrankenPHP in the Joomla Community and other "Super Joomlers" would like to make an article about FrankenPHP in the official Joomla Community Magazine (JCM). I don't like too much to write articles. Audio format is my favourite format. But I don't have a podcast yet.

**Reply by alexandreelise** [reply]
My joomler friends developers and non-developers find it cool and useful. Hopefully the word will spread that PHP and Golang are now best friends thanks to FrankenPHP.

**Reply by alexandreelise** [reply]
Hi Kévin @dunglas,

Here is a reproducer Dockerfile that shed a light on what seems to be an issue:

```Dockerfile
FROM dunglas/frankenphp

RUN mkdir -p /usr/src/external/{a,b,c} \
    && echo '<?php echo __DIR__; ' > /usr/src/external/{a,b,c}/index.php \
    && ln -s /usr/src/external/a /app/public/a \
    && ln -s /usr/src/external/b /app/public/b \
    && ln -s /usr/src/external/c /app/public/c

CMD frankenphp run --config /etc/caddy/Caddyfile
```

**Reply by alexandreelise** [reply]
Salut Kévin @dunglas, Je crois avoir trouvé une piste.

https://gitlab.alpinelinux.org/alpine/apk-tools/-/issues/9634

Passe un excellent weekend!

**Reply by alexandreelise** [reply]
Wow! I just realized that dunglas/frankenphp image is based on Debian 12.2 which makes my previous comment totally irrelevant. Enjoy your weekend anyway.

**Reply by alexandreelise** [reply]
Edited reproducer that seems to behave correctly and shows the resolved symlinks using realpath in php

```Dockerfile

FROM dunglas/frankenphp


RUN [ "/bin/bash", "-c", "mkdir -p /usr/src/external/{a,b,c} \
    && echo '<?php echo realpath(__DIR__);' | tee /usr/src/external/{a,b,c}/index.php \
    && ln -s /usr/src/external/a /app/public/a \
    && ln -s /usr/src/external/b /app/public/b \
    && ln -s /usr/src/external/c /app/public/c" ]

VOLUME /app/public

CMD frankenphp run --config /etc/caddy/Caddyfile

```

NOTE: The exec form of RUN is used, in order to force the use of bash shell and allow brace expansion. Otherwise it fails miserably.

**Reply by alexandreelise** [reply]
Hi, there is some progress. After reading caddy server docs I found out that I needed to customize my Caddyfile to make it work with FrankenPHP and Joomla. Still not there yet for want I want to do. But hopefully will get there soon. Stay tuned!

**Reply by alexandreelise** [reply]
Now auto-install is a reality! At last! https://github.com/alexandreelise/frankenphp-joomla/pull/3

**Reply by alexandreelise** [reply]
Thanks for your support anyway
